tasks/pkg_template.py

```diff
@@ -0,0 +1,78 @@
+import os
+import shutil
+import sys
+from contextlib import chdir
+
+from invoke.context import Context
+from invoke.tasks import task
+
+
+@task
+def generate(ctx: Context):
+    """
+    Generate the code for the template package.
+    Takes the code from the Go standard library and applies the patches.
+    """
+    if not shutil.which("gopatch"):
+        print("gopatch could not be found, you need to run `inv -e install-tools` first...")
+        sys.exit(1)
+
+    with chdir("pkg/template"):
+        _generate(ctx)
+
+
+def _generate(ctx: Context):
+    print(f"Generating code for Go version {get_go_version(ctx)}")
+
+    goroot = get_goroot(ctx)
+    if not goroot:
+        print("Could not find Go's source code path")
+        sys.exit(1)
+    print(f"Using code from {goroot}")
+
+    print("Removing previous code...")
+    remove_dirs = ["text", "html", "internal"]
+    for dir_name in remove_dirs:
+        if os.path.exists(dir_name):
+            shutil.rmtree(dir_name)
```

Comment:
Should the deletion be in the `pkg/template/..`  folder?

Reply:
Same for the creation of directories 😄

Reply:
Yeah that's the case, the current directory is `pkg/template` (there is a `chdir` at the top)

Reply:
It avoids having to prefix every file with `pkg/template` or doing `ctx.cd`

Reply:
Nevermind I just saw `chdir("pkg/template"):` 🤦

Reply:
Should we move the `chdir("pkg/template"):` inside generate?

Reply:
The whole function would have to run in the `with` block, that's why everything is in a `_generate` function, it avoids one indentation level everywhere